### ISSUE TITLE
[Feature]: Save settings to directory

### ISSUE THREAD

**Issue by StrollStars** [sqlitebrowser/sqlitebrowser#3045]
### Describe the new feature

I hope to save settings to the directory where sqlitebrowser is or I choose.

**Comment by lucydodo** [comment]
Thanks for good suggestion. I've implemented this feature before. (#2390, #2420)
However, I've confirmed that this feature doesn't work for now. I will fix it ASAP.

**Comment by uvtofu** [comment]
Thanks for response and your code contribution.

I have read your mentioned posts ( #2390, #2420), and had a feeling that it MUST uses a command line flag.
`>> sqlitebrowser --settings <setting_file>`
So in portable case, you should remember the flag, open console window and input the command line.
(or create a custom batch/shell script to avoid command line everytime.)
In local install, you should edit desktop shortcut (.lnk for windows, .desktop for linux).
Is this right?

Blender uses a simpler approach.
You create portable config folder in place, and forget it.
Since then, Blender automatically uses the portable config.
There is no bother to handle command line or edit desktop shortcut, in user's side.

Could you give Blender a try?

**Comment by lucydodo** [comment]
@uvtofu That's a good point. So, what do you think about design a 'configuration.ini' file to be loaded if it exists?

**Comment by uvtofu** [comment]
[lucydodo](https://github.com/lucydodo)
IMHO, the portable config file should better use same filename as non-portable one, this add convinience when user migrate between portable and non-portable use cases.
I also suggest you consider a config folder, the reasons are,
- it is easier to create a folder than a file in a GUI file browser.
- It is easier for expansion, in case that DB4S would adopt setting-backup or multiple-settings mechanism in the future.

As for the filename/foldername, From common sense, it should be easy to remember (english word not too long), and case-sensitive in POSIX systems. In case of single config file, the filename prefix should better be same as the main executable.
I think DB4S linux version uses "sqlitebrowser.conf" is a good file name. I run dual-boot computer (win/linux), so I always welcome cross-os apps could share their config file between systems.

Thank you again, and hope more people contribute ideas.

**Comment by lucydodo** [comment]
@uvtofu If so, how about the DB4S to load the 'sqlitebrowser.conf' file if it exists in the 'config' directory?

**Comment by uvtofu** [comment]
@lucydodo 
I still prefer use the 'config' folder as sole condition, like Blender does.
If 'config' folder exists in executable directory, DB4S enter portable mode, no matter 'sqlitebrowser.conf' exist or not under 'config' .

My pseudo code below,

```
if ./config exist:
    if ./config/sqlitebrowser.conf exist:
        then usefile ./conifg/sqlitebrowser.conf
    else:
        makefile ./config/sqlitebrowser.conf
        usefile ./config/sqlitebrowser.conf

else:
    <make/use non-portable config file>

```

**Comment by sky5walk** [comment]
Interesting stuff.
If the goals are cross platform and portable, please drop the Windows registry writes entirely.
I have been quite frustrated with DB4S lack of portability and criss-crossed settings on Windows installations.

Firstly, define the settings within a reasonably simple SQLite database schema.
The schema must support multiple versions to allow user selection and undo's.

I care little if the settings.db is in a sub-folder or at the root of installation.
But, one less folder and path to manage is simpler.

**Comment by lucydodo** [comment]
I made a PR by code that can solve this issue.
It will be reflected in the 'master' branch after code review.

That PR make DB4S will load settings if it has either of these two files:
1. `./configuration/config.conf`
2. `./configuration/config.ini`

If you have other thoughts, please leave a comments. Thanks. 😄

**Comment by chrisjlocke** [comment]
> The schema must support multiple versions to allow user selection and undo's.

How we do it at work is having 'profiles' - similar to how Visual Studio works, with 'debug' and 'live' configurations, for example.  But rather than having 'fonts/main window font size' as an option, we just prefix that with a profile, so 'main/fonts/main window font size', so we can easily switch profiles.  The main option is therefore which profile you want.

**Comment by mgrojo** [comment]
This still requires the user to know the functionality and create the file somehow, doesn't it? I think the 100% portable mode should be controlled through installation. The portable packaging should create a desktop entry that uses the command line to save the settings to a file.

**Comment by lucydodo** [comment]
I also agree with @mgrojo. To add a bit more comments, I think it would be better if the installer creates a shortcut to the executable file containing the command line to call the configuration file, and the location of the configuration file is located in the 'configuration' folder so that the user can easily find it. . What do other people think? 😄

**Comment by chrisjlocke** [comment]
> This still requires the user to know the functionality and create the file somehow, doesn't it?
This is provided in the settings screen - 'export configuration'.  The user can store this file where they want.  Unfortuately, they can also call it what they want.

If they store the file with the .exe (or in the 'configuration' directory) then DB4S uses this by default - this negates the need to create shortcuts, etc, yes?

**Comment by lucydodo** [comment]
DB4S has the advantage of having fewer condition comparison commands when creating shortcuts that contain the portable installer command line. (Of course, the overhead given by this condition is not significant.) 😄

**Comment by MaDill** [comment]
Sorry to warm up this old issue. Is this implemented? I use version 3.13.1 and created a folder configuration and the file config.ini inside. The ini file stays empty and the values are still written into the windows registry. Did I made something wrong? I was expecting that all settings now goes to the config.ini and not anymore into the registry.
